### ISSUE TITLE
[Bug]: Issue with BiDi-powered logging

### ISSUE THREAD

**Issue by mathiasbynens** [puppeteer/puppeteer#10032]
### Bug expectation

I expected to get a message logged to my terminal, but I didn’t get any.

### Bug behavior

- [ ] Flaky
- [ ] PDF

### Minimal, reproducible example

```TypeScript
import puppeteer from 'puppeteer';

const browser = await puppeteer.launch({
  protocol: 'webDriverBiDi',
});

const context = await browser.createIncognitoBrowserContext();
const page = await context.newPage();

page.on('console', (message) => {
  console.log(`Unexpected console message received: [${
    message.type()
  }] ${message.text()}`);
});


await page.goto(`data:text/html,<script>console.log('1');</script>`);
```

### Error string

No error

### Puppeteer version

19.9.0

### Node version

16.14.0

### Package manager

npm

### Package manager version

8.14.0

### Operating system

macOS

**Comment by sadym-chromium** [comment]
I added the following script and run it. I get the `Unexpected console message received` message in console. What did I do wrong? 
```
const puppeteer = require('puppeteer');

(async () => {

  const browser = await puppeteer.launch({
    protocol: 'webDriverBiDi',
  });

  const context = await browser.createIncognitoBrowserContext();
  const page = await context.newPage();

  page.on('console', (message) => {
    console.log(`Unexpected console message received: [${
      message.type()
    }] ${message.text()}`);
  });

  await page.goto(`data:text/html,<script>console.log('1');</script>`);
})();
```

**Comment by sadym-chromium** [comment]
First suspect is not waiting for the subscribe command to finish: https://github.com/puppeteer/puppeteer/blob/57c57274243e3e83f1fe8f8fdb37eb8d3911d32d/packages/puppeteer-core/src/common/bidi/Page.ts#L53

**Comment by Lightning00Blade** [comment]
Closed by #10060
